core/controller/src/main/scala/whisk/core/controller/Actions.scala

```diff
@@ -336,40 +340,103 @@ trait WhiskActionsApi extends WhiskCollectionAPI {
+                        makeWhiskAction(fixedContent, namespace, name)
+                    } recoverWith{
+                        case _: TooManyActionsInSequence =>
+                            Future failed RejectRequest(BadRequest, "too many actions in sequence")
+                        case _: SequenceWithRecursion =>
+                            Future failed RejectRequest(BadRequest, "recursion detected in sequence")
+                        case _: NoDocumentException =>
+                            Future failed RejectRequest(BadRequest, "action not found")
+                        // anything else that should be caught here?
+                    }
+                case _ => Future successful {makeWhiskAction(content, namespace, name)}
             }
-
-            WhiskAction(
-                namespace,
-                name,
-                exec,
-                parameters,
-                limits,
-                content.version getOrElse SemVer(),
-                content.publish getOrElse false,
-                content.annotations getOrElse Parameters())
         }
         else Future failed RejectRequest(BadRequest, "exec undefined")
     }
 
+    /**
+     * helper function that creates a whisk action
+     */
+    private def makeWhiskAction(content: WhiskActionPut, namespace: EntityPath, name: EntityName)(implicit transid: TransactionId) = {
+        val exec = content.exec.get
+        val limits = content.limits map { l =>
+                        ActionLimits(
+                            l.timeout getOrElse TimeLimit(),
+                            l.memory getOrElse MemoryLimit(),
+                            l.logs getOrElse LogLimit())
+                    } getOrElse ActionLimits()
+        // This is temporary while we are making sequencing directly supported in the controller.
+        // The parameter override allows this to work with Pipecode.code. Any parameters other
+        // than the action sequence itself are discarded and have no effect.
+        val parameters = exec match {
+            case seq: SequenceExec => Parameters("_actions", JsArray(seq.components map { JsString(_) }))
+            case _                 => content.parameters getOrElse Parameters()
+        }
+
+        WhiskAction(
+            namespace,
+            name,
+            exec,
+            parameters,
+            limits,
+            content.version getOrElse SemVer(),
+            content.publish getOrElse false,
+            content.annotations getOrElse Parameters())
+    }
+
     /** Updates a WhiskAction from PUT content, merging old action where necessary. */
-    private def update(content: WhiskActionPut)(action: WhiskAction) = Future successful {
+    private def update(user: Identity, content: WhiskActionPut)(action: WhiskAction)(implicit transid: TransactionId) = {
+        if (content.exec.isDefined) {
```

Comment:
Would it make sense to use the more idiomatic `content.exec.map { // do something when exec is Some }.getOrElse { // do something when exec is None }`